DFC.ServiceTaxonomy.GraphSync/GraphSyncers/Fields/ContentPickerFieldGraphSyncer.cs

```diff
@@ -196,9 +249,14 @@ private async Task<ContentItem[]> GetLatestContentItemsFromIds(JArray contentIte
         }
 
         private object GetNodeId(ContentItem pickedContentItem, IGraphMergeContext context)
-        {
+        {  
+            context.SyncNameProvider.ContentType = pickedContentItem.ContentType;
```

Comment:
We mustn't set `ContentType` on the `SyncNameProvider` in the context, as subsequent syncers assume it's set for the content item it's syncing. (We should probably change `SyncNameProvider` to stop the `ContentType` being set again.)

To get the GraphSyncPartSettings, we could make GetGraphSyncPartSettings public and add it to `ISyncNameProvider`. (ISyncNameProvider has 2 sets of methods, those that need `ContentType` setting first, and those that accept a `ContentType`.)